cdap-ui/app/cdap/components/Experiments/store/ExperimentDetailActionCreator.js

```diff
@@ -207,134 +155,134 @@ function getSplitsInExperiment(experimentId) {
           splits
         }
       });
+    }, (err) => {
+      setExperimentDetailError(`Failed to get splits in experiment '${experimentId}': ${err.response || err}`);
     });
 }
 
-function getModelStatus(experimentId, modelId) {
+function pollModelStatus(experimentId, modelId) {
   return myExperimentsApi
     .pollModelStatus({
       namespace: getCurrentNamespace(),
       experimentId,
       modelId
     })
     .subscribe(modelStatus => {
+      removeModelsWithError(modelId);
       experimentDetailsStore.dispatch({
         type: EXPERIMENTDETAILACTIONS.SET_MODEL_STATUS,
         payload: {
           modelId,
           modelStatus
         }
       });
+    }, () => {
+      addModelsWithError(modelId);
     });
 }
 
-function setActiveModel(activeModelId) {
-  let state = experimentDetailsStore.getState();
-  let {splitDetails} = state.models.find(model => model.id === activeModelId);
-  if (!splitDetails) {
-    getSplitsInExperiment(state.name);
+function getModelStatus(experimentId, modelId) {
+  addModelsLoading(modelId);
+  myExperimentsApi
+    .getModelStatus({
+      namespace: getCurrentNamespace(),
+      experimentId,
+      modelId
+    })
+    .subscribe(modelStatus => {
+      removeModelsWithError(modelId);
+      experimentDetailsStore.dispatch({
+        type: EXPERIMENTDETAILACTIONS.SET_MODEL_STATUS,
+        payload: {
+          modelId,
+          modelStatus
+        }
+      });
+    }, () => {
+      addModelsWithError(modelId);
+    }, () => {
+      removeModelsLoading(modelId);
```

Comment:
Do we need the `onComplete` function handler? Won't we always hit the `onSuccess` or `onError` callbacks?

Reply:
Yes, so this is so we don't have to repeat the `removeModelsLoading(modelId);` for both the success and error callbacks. This way `removeModelsLoading(modelId);` will always be called no matter if the request succeeded or failed.